src/services/lens.ts

```diff
@@ -69,7 +59,8 @@ export class LensService<T extends ChainId> extends ContractService<T> {
    * @returns list of assets
    */
   async getAssets(overrides: CallOverrides = {}): Promise<GenericAsset[]> {
-    return await this.contract.read.getAssets(overrides).then(structArray);
+    let contract = await this.contract;
+    return contract.read.getAssets(overrides).then(structArray);
```

Comment:
All these changes should just be
```suggestion
    return this.contract.read.getAssets(overrides).then(structArray);
```

Reply:
`this.contract` becomes a promise in this PR, we can't call `read` on it unless it resolves first.

Reply:
Ahh! Ok, that makes sense then